adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java

```diff
@@ -447,4 +445,108 @@ void addFareRuleAndGetFareRuleShouldReturnSameEntity() {
+
+    @Test
+    void addTwiceSameShapePointShouldReturnNull() {
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+        when(mockShapePoint.getShapeId()).thenReturn("test id");
+        when(mockShapePoint.getShapePtLat()).thenReturn(50f);
+        when(mockShapePoint.getShapePtLon()).thenReturn(100f);
+        when(mockShapePoint.getShapePtSequence()).thenReturn(4);
+        when(mockShapePoint.getShapeDistTraveled()).thenReturn(56f);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        underTest.addShapePoint(mockShapePoint);
+        assertNull(underTest.addShapePoint(mockShapePoint));
+    }
+
+    @Test
+    void addShapePointWithSameDataShouldReturnNull () {
+        final ShapePoint firstShapePoint = mock(ShapePoint.class);
+        when(firstShapePoint.getShapeId()).thenReturn("test id00");
+        when(firstShapePoint.getShapePtLat()).thenReturn(50f);
+        when(firstShapePoint.getShapePtLon()).thenReturn(100f);
+        when(firstShapePoint.getShapePtSequence()).thenReturn(4);
+        when(firstShapePoint.getShapeDistTraveled()).thenReturn(56f);
+
+        final ShapePoint duplicateShapePoint = mock(ShapePoint.class);
+        when(duplicateShapePoint.getShapeId()).thenReturn("test id00");
+        when(duplicateShapePoint.getShapePtLat()).thenReturn(50f);
+        when(duplicateShapePoint.getShapePtLon()).thenReturn(100f);
+        when(duplicateShapePoint.getShapePtSequence()).thenReturn(4);
+        when(duplicateShapePoint.getShapeDistTraveled()).thenReturn(56f);
+
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        assertEquals(firstShapePoint, underTest.addShapePoint(firstShapePoint));
+        assertNull(underTest.addShapePoint(duplicateShapePoint));
+    }
+
+    @Test
+    void addNullShapePointShouldThrowException() {
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        final Exception exception = assertThrows(IllegalArgumentException.class,
+                () -> underTest.addShapePoint(null));
+        assertEquals("Cannot add null shape point to data repository", exception.getMessage());
+    }
+
+    @Test
+    void addShapeAndGetShapeByIdShouldReturnSameEntity() {
+        final ShapePoint mockShapePoint00 = mock(ShapePoint.class);
+        when(mockShapePoint00.getShapeId()).thenReturn("test id00");
+        when(mockShapePoint00.getShapePtSequence()).thenReturn(4);
+
+        final ShapePoint mockShapePoint01 = mock(ShapePoint.class);
+        when(mockShapePoint01.getShapeId()).thenReturn("test id01");
+        when(mockShapePoint01.getShapePtSequence()).thenReturn(8);
+
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        underTest.addShapePoint(mockShapePoint00);
+        underTest.addShapePoint(mockShapePoint01);
+
+        final TreeMap<Integer, ShapePoint> firstMapToCheck = new TreeMap<>();
```

Comment:
Shouldn't this use the repo instead of instatiating your own TreeMaps here?

Reply:
Absolutely, I missed that point. Thank you!